source/client-side-encryption/client-side-encryption.rst

```diff
@@ -286,13 +289,86 @@ Data keys are stored as documents in a special MongoDB collection. Data
 keys are protected with encryption by a KMS provider (AWS KMS, Azure key
 vault, GCP KMS, or a local master key).
 
+metadataClient
```

Comment:
I get why we expose the `metadataClient` option, but I think this needs some additional language to guard against (very) invalid use-cases. Consider the following snippet:

```
metadataClient = new Client("mongodb://server1:27017", minPoolSize=2);
client = new Client("mongodb://server2:27017", maxPoolSize=1, metadataClient);
```

The fact that users would be able to direct the metadata client to a separate server concerns me. Depending on how feasible it is to guard against this in different drivers, we may want to consider exposing only a `metadataClientOptions` options that allows overriding options used for constructing the internal metadata client instead of allowing to specify a client object. This would allow users to do this:
```
client = new Client("mongodb://server2:27017", maxPoolSize=1, metadataClientOptions={minPoolSize=2, maxPoolSize=5})
```

Reply:
This could be a big headache for Java and .NET drivers.  Both of them use a MongoClientSettings class that encompasses optional values and required values like credentials and hosts.  Would this require duplicating that whole structure into one that included everything but the hosts and credentials?

Reply:
Yes I think so. Is that any harder than copying a `MongoClientSettings` and excluding `AutoEncyptionOpts`? That's already necessary for the current proposed change to configure the internal client.

Reply:
I think that's different.  In that case we're just copying the values from one MongoClientSettings into another, and excluding AutoEntrypOps.  in this case, do drivers have to provide a public API that excludes the possibility of configuring different hosts?  Or could the driver just ignore the configured hosts? Or am I misunderstanding this completely?

Reply:
I see what you mean.

Ignoring the configured `hosts` and `credentials` seems like a weird API that ignores a core part of the `MongoClientSettings`.

We want to expose an option that:
1. Let's users completely configure the internal `MongoClient` (notably to configure the connection pool sizes, since `listCollections` / `find` on the key vault will likely be much less frequent then normal `MongoClient` operations)
2. Does not have ignored options
3. Does not permit misuse (via connecting to the wrong cluster)

If that requires non-trivial driver changes, I am in favor of not exposing any configuration, and instead choosing sensible defaults.

If we need an internal client, copy the `MongoClient` options. Exclude `AutoEncryptionOpts`. Set `minPoolSize` to 0.

It's one less knob. And if we determine that we need additional configuration later, we can add it.

Reply:
68374f8 proposes not exposing `metadataClient`.